lib/plugins/platform/platform.js

```diff
@@ -65,14 +65,36 @@ class Platform {
     if (this.provider) {
       this.hooks = {
         'after:deploy:deploy': this.publishService.bind(this),
-        'after:remove:remove': this.removeService.bind(this),
+        'after:remove:remove': this.archiveService.bind(this),
       };
     }
   }
 
-  removeService() {
-    // TODO implement platform removal here
-    return BbPromise.resolve();
+  archiveService() {
+    const authToken = this.getAuthToken();
+    const publishFlag = selectServicePublish(this.serverless.service);
+    if (!authToken || !publishFlag) {
+      // NOTE removeService is an opt-in feature and no warning is needed
+      return BbPromise.resolve();
+    }
+
+    if (authToken && authToken.length > 8000) {
```

Comment:
Is this `8000` value a typical value for such a scenario? Is this a timeout in ms?

Reply:
I followed the pattern in [`publishService`](https://github.com/serverless/serverless/pull/4256/files#diff-058c83b143cc177d962694d54fb23e4fL105).

Reply:
This is actually the length of [`id_token` stored in localStorage](https://github.com/serverless/serverless/blob/e29b633275dbbd3037eafae2bb990b33bf364272/lib/utils/getAuthToken.js#L14)

Reply:
I thought the token keeps the same lenght. If that's the case this snippet doesn't make sense to me. We added it here: https://github.com/serverless/serverless/commit/db02e800

I reached out to David for more details and get back to you asap.

Reply:
Thanks for the clarification 👍